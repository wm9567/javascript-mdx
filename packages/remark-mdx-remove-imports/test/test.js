const fs = require('fs')
const path = require('path')
const unified = require('unified')
const remarkParse = require('remark-parse')
const remarkStringify = require('remark-stringify')
const remarkMdx = require('remark-mdx')

const removeImports = require('..')

const fixture = fs.readFileSync(path.join(__dirname, './fixture.md'), 'utf8')

const stringify = mdx => {
  const result = unified()
    .use(remarkParse)
    .use(remarkStringify)
    .use(remarkMdx)
    .use(removeImports)
    .processSync(mdx)

  return result.contents
}

it('removes imports', () => {
  const result = stringify(fixture)

  expect(result).toMatchSnapshot()
})
