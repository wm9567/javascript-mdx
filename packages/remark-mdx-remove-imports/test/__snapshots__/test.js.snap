// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removes imports 1`] = `
"export default props => <div {...props} />

# Hello, world!

This is a paragraph

\`\`\`js
const foo = 'bar'
\`\`\`
"
`;
